lang/java/avro/src/main/java/org/apache/avro/io/parsing/ResolvingGrammarGenerator.java

```diff
@@ -113,8 +113,7 @@ private Symbol generate(Resolver.Action action, Map<Object, Symbol> seen) throws
       }
       return Symbol.seq(Symbol.alt(symbols, labels), Symbol.WRITER_UNION_ACTION);
 
-    }
-    if (action instanceof Resolver.ReaderUnion) {
+    } else if (action instanceof Resolver.ReaderUnion) {
```

Comment:
No functional change here, but it was impossible to fall through to this line if any of the above conditions matched -- they all always return. This just makes that more clear.